genesyscloud/integration_credential/resource_genesyscloud_integration_credential.go

```diff
@@ -82,16 +83,24 @@ func getAllCredentials(ctx context.Context, clientConfig *platformclientv2.Confi
 func createCredential(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	name := d.Get("name").(string)
 	cred_type := d.Get("credential_type_name").(string)
+	fields := buildCredentialFields(d)
+	_, secretFieldPresent := fields["clientSecret"]
 
 	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
 	ip := getIntegrationCredsProxy(sdkConfig)
 
+	//If if is a Genesys Cloud OAuth Client and the user has not provided a secret field we should look for the
+	//item in the cache DEVTOOLING-448
+	if cred_type == "pureCloudOAuthClient" && !secretFieldPresent {
```

Comment:
Here is where we check to see if we are dealing with a Genesys Cloud AuthClient and whether the user has set the secret field has been set.  If the secretField has not been set, then the code will try to retrieve an OAuthClient  based on the client id from the the oauthClient proxy and see if there is already an OauthClient present in the code.